tests/components/home_connect/test_sensor.py

```diff
@@ -509,143 +555,150 @@ async def test_remaining_prog_time_edge_cases(
+    event_key: EventKey,
+    value_expected_state: list[tuple[str, str]],
+    appliance: HomeAppliance,
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[MagicMock], Awaitable[bool]],
+    setup_credentials: None,
+    client: MagicMock,
+) -> None:
+    """Tests for appliance sensors."""
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    assert await integration_setup(client)
+    assert config_entry.state == ConfigEntryState.LOADED
+
+    for value, expected_state in value_expected_state:
+        await client.add_events(
+            [
+                EventMessage(
+                    appliance.ha_id,
+                    EventType.STATUS,
+                    ArrayOfEvents(
+                        [
+                            Event(
+                                key=event_key,
+                                raw_key=str(event_key),
+                                timestamp=0,
+                                level="",
+                                handling="",
+                                value=value,
+                            )
+                        ],
+                    ),
+                ),
+            ]
+        )
+        await hass.async_block_till_done()
+        assert hass.states.is_state(entity_id, expected_state)
+
+
+@pytest.mark.parametrize(
+    (
+        "entity_id",
+        "event_key",
+        "appliance",
+    ),
+    [
         (
             "sensor.fridgefreezer_freezer_door_alarm",
             EventKey.REFRIGERATION_FRIDGE_FREEZER_EVENT_DOOR_ALARM_FREEZER,
-            EventType.EVENT,
-            BSH_EVENT_PRESENT_STATE_CONFIRMED,
-            "confirmed",
             "FridgeFreezer",
         ),
-        (
-            "sensor.coffeemaker_bean_container_empty",
-            EventType.EVENT,
-            "EVENT_NOT_IN_STATUS_YET_SO_SET_TO_OFF",
-            "",
-            "off",
-            "CoffeeMaker",
-        ),
         (
             "sensor.coffeemaker_bean_container_empty",
             EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_BEAN_CONTAINER_EMPTY,
-            EventType.EVENT,
-            BSH_EVENT_PRESENT_STATE_OFF,
-            "off",
-            "CoffeeMaker",
-        ),
-        (
-            "sensor.coffeemaker_bean_container_empty",
-            EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_BEAN_CONTAINER_EMPTY,
-            EventType.EVENT,
-            BSH_EVENT_PRESENT_STATE_PRESENT,
-            "present",
-            "CoffeeMaker",
-        ),
-        (
-            "sensor.coffeemaker_bean_container_empty",
-            EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_BEAN_CONTAINER_EMPTY,
-            EventType.EVENT,
-            BSH_EVENT_PRESENT_STATE_CONFIRMED,
-            "confirmed",
             "CoffeeMaker",
         ),
     ],
     indirect=["appliance"],
 )
-async def test_sensors_states(
+@patch(
+    "homeassistant.components.home_connect.sensor._add_event_sensor_entity",
```

Comment:
It's enough to assert the entity state in the state machine. We can remove this.

Reply:
Isn't interesting to assert that it is only called once?
https://github.com/home-assistant/core/blob/d6db6d291e0a087c649e7596354139a143cf267d/tests/components/home_connect/test_sensor.py#L667-L669
https://github.com/home-assistant/core/blob/d6db6d291e0a087c649e7596354139a143cf267d/tests/components/home_connect/test_sensor.py#L698-L701

Reply:
No, we don't want to assert integration details.

Reply:
Then how do we check that the listener is correctly deleted after adding the entity?
Maybe in the future someone breaks this functionality and the integration attempts to add an entity every time the event is received.

Reply:
It should not add extra entities. That would raise an error for duplicate unique ids. That we can test, ie just send the event again, and check that still a single entity exists and without problems.

Reply:
Well, I tried to not delete the event, so that the entity is attempted to be added multiple times. It doesn't raise any error and the `add_to_platform_abort` method is called. A single entity exists and without problems. only error logs. So we cannot detect that kind of issues with the actual APIs.

So if you want to remove that assertion, we can remove it, but we cannot test that the event is deleted. And the only way we will know that the mentioned issue is happening is through logs...

If you are okey with that, I will delete that assertion

Reply:
Yes, we should remove this assertion.

We can assert logs.

Reply:
> We can assert logs

Any example about how to do it?
Or can I mock the `entity_platform.py`'s logger?